engine/collection/compliance/engine.go

```diff
@@ -65,16 +62,16 @@ func NewEngine(
 	}
 
 	eng := &Engine{
-		log:                        engineLog,
-		metrics:                    core.engineMetrics,
-		me:                         me,
-		headers:                    core.headers,
-		payloads:                   payloads,
-		state:                      state,
-		core:                       core,
-		pendingBlocks:              blocksQueue,
-		pendingBlocksNotifier:      engine.NewNotifier(),
-		finalizationEventsNotifier: engine.NewNotifier(),
+		log:                          engineLog,
+		metrics:                      core.engineMetrics,
+		me:                           me,
+		headers:                      core.headers,
+		payloads:                     payloads,
+		state:                        state,
+		core:                         core,
+		pendingBlocks:                blocksQueue,
+		pendingBlocksNotifier:        engine.NewNotifier(),
+		finalizedBlocksNotifications: make(chan *model.Block, 10),
```

Comment:
## ⚠️ 
From a conceptual standpoint, I dislike the usage of a buffered channel here, especially of size 10. When there is a higher fraction of offline primaries, I believe it is not too unlikely that finalization could halt for more than 10 views. (see probability approximation here in [this notion doc](https://www.notion.so/dapperlabs/Probability-of-finalization-halt-for-n-views-72997a3bff2446ae96e1824e57eb04de))
After such a temporary finalization halt, there will be a quick burst of finalized blocks, once we have a sufficient sequence of 2 subsequent honest primaries. If we exceed the capacity of 10 finalized blocks, the compliance engine will start dropping the newer finalized blocks and therefore delay pruning. 


### Suggestion
I believe there is not really any reason to work with a channel here. In fact, we are only interested in the _latest_ finalized block (by view). We have high-performance, completely non-blocking implementations [here](https://github.com/onflow/flow-go/blob/b4a7dbb4a6b7cf09a481f7eaa49c5bd8057f2427/consensus/hotstuff/tracker/tracker.go) for tracking the latest QC and TC. It is really not a lot of code. 

I would suggest to add a similar Tracker for latest block by view. I believe this would be widely useful beyond this PR.

Reply:
Done